### PR TITLE
feat: repository configuration

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Repository initialization:

configured repository to build a docker image with an example function app written in kotlin

#### Motivation and Context
Repository pagopa-ecommerce-transactions-functions first configuration

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by giovanniberti — DISMISSED** [review]
🚀

File: README.md [hunk 1]
```diff
@@ -1,42 +1,58 @@
...
-`mvn clean package`
+`mvn validate` used to perform ecommerce-commons library checkout from git repo and install throught maven plugin
 
-`mvn azure-functions:run`
+`mvn clean package azure-functions:run` perform clean and package maven scopes and then starts the functions locally
 
-### Test
-`curl http://localhost:7071/example` 
+For testing purpose the commons reference can be change from a specific release to a branch by changing the following
+configurations tags:
```
Comment (giovanniberti) [inline README.md:25]:
```suggestion
configurations tags inside `pom.xml`:
```

File: README.md [hunk 2]
```diff
@@ -1,42 +1,58 @@
...
 
----
+FROM:
+
+```sh
```
Comment (giovanniberti) [inline README.md:29]:
```suggestion
```xml
```

File: pom.xml [hunk 3]
```diff
@@ -3,20 +3,26 @@
...
 
-    <name>Azure Custom Fn</name>
+    <name>pagopa-ecommerce-transactions-functions</name>
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version>11</java.version>
+        <java.version>17</java.version>
         <azure.functions.maven.plugin.version>1.15.0</azure.functions.maven.plugin.version>
         <azure.functions.java.library.version>1.4.2</azure.functions.java.library.version>
```
Comment (infantesimone) [inline pom.xml:17]: is it used?

File: pom.xml [hunk 4]
```diff
@@ -134,6 +146,42 @@
             <version>2.13.1</version>
         </dependency>
 
+        <dependency>
+            <groupId>it.pagopa</groupId>
+            <artifactId>pagopa-ecommerce-commons</artifactId>
+            <version>${pagopa-ecommerce-commons.version}</version>
+            <exclusions>
```
Comment (infantesimone) [inline pom.xml:153]: So in ecommerce common we import all this deps?

File: README.md (hunk 1 above)
Comment (pietro-tota) [reply README.md]: Fixed in https://github.com/pagopa/pagopa-ecommerce-transactions-functions/pull/2/commits/62ff26f0f1dd6a1927d956d83aa648b3a17455e2

File: README.md (hunk 2 above)
Comment (pietro-tota) [reply README.md]: Fixed in https://github.com/pagopa/pagopa-ecommerce-transactions-functions/pull/2/commits/62ff26f0f1dd6a1927d956d83aa648b3a17455e2

File: pom.xml (hunk 4 above)
Comment (pietro-tota) [reply pom.xml:153]: Yes, now it's used in place of the constant value in the pom. see https://github.com/pagopa/pagopa-ecommerce-transactions-functions/pull/2/commits/62ff26f0f1dd6a1927d956d83aa648b3a17455e2

File: pom.xml (hunk 3 above)
Comment (pietro-tota) [reply pom.xml:17]: Yes, now it's used in place of the constant value in the pom. see https://github.com/pagopa/pagopa-ecommerce-transactions-functions/commit/62ff26f0f1dd6a1927d956d83aa648b3a17455e2

File: pom.xml (hunk 4 above)
Comment (giovanniberti) [reply pom.xml:153]: They are used inside ecommerce-commons for repositories (and related bells and whistles like automatic conversion/serialization). My Maven-fu it's not that advanced but if I had to guess removing those dependencies would probably break the commons build.
A solution could be to split commons into [two modules](https://www.baeldung.com/maven-multi-module) (something like `core`/`data` and `spring`?).

Comment (infantesimone) [reply pom.xml:154]: good solution to do in the next sprints :-)

File: Dockerfile [hunk 5]
```diff
@@ -1,22 +1,28 @@
...
+    ./mvnw clean package -DskipTests && \
     cd ./target/azure-functions/ && \
     cd $(ls -d */|head -n 1) && \
     cp -a . /home/site/wwwroot
 
-# This image is ssh enabled
-#FROM mcr.microsoft.com/azure-functions/java:3.0-java$JAVA_VERSION-appservice
-# This image isn't ssh enabled
-FROM mcr.microsoft.com/azure-functions/java:3.0-java$JAVA_VERSION
+FROM mcr.microsoft.com/azure-functions/java:4-java$JAVA_VERSION
```
Comment (infantesimone) [inline Dockerfile:22]: why not use `4-java17-slim`?

Comment (infantesimone) [reply Dockerfile:22]: so we will have a lighter container running, and leave the other image only for the build phase.

Comment (pietro-tota) [reply Dockerfile]: Fixed in https://github.com/pagopa/pagopa-ecommerce-transactions-functions/pull/2/commits/9200277f881d14657d5ff13c01371aa9d722886b

**Review by infantesimone — APPROVED** [review]

**Pull request merged**